ATM/ATM.py

```diff
@@ -0,0 +1,164 @@
+    while number_of_wrong_pin_entries < 3:
+        pin_entry = input("Please enter your four digit PIN ")
+        if pin_entry == pin:
+            return True
+        else:
+            number_of_wrong_pin_entries += 1
+            print("Number of wrong pin entries " + str(number_of_wrong_pin_entries))
+    print("You have entered a wrong PIN three times. Your card has been blocked. Have a nice day.")
+    quit()
+
+
+def set_initial_balance():
+    balance = input("Please enter your initial balance. ")
+    while not is_input_a_positive_float_with_a_max_of_two_decimal_places_check(balance):
+        balance = input("Please enter your initial balance. ")
+    print("Thank you.")
+    time.sleep(1)
+    return balance
+
+
+def is_input_a_positive_float_with_a_max_of_two_decimal_places_check(input_to_be_checked):
+    try:
+        float(input_to_be_checked)
+    except:
+        print("You entered characters other than digits. Please try again.")
+        return False
+
+    if float(input_to_be_checked) <= 0:
+        print("Please enter a positive amount.")
+        return False
+
+    if "." in input_to_be_checked:
+        if len((input_to_be_checked.split("."))[1]) > 2:
+            print("Please enter an amount without fractions of cents.")
+            return False
+        else:
+            return True
+    else:
+        return True
+
+
+def withdraw():
+    amount_chosen = input("How much money would you like to withdraw? You may withdraw 20 EUR, 50 EUR or (O)ther amount.")
+
+    while not amount_chosen in ["20", "50", "O"]:
+        print("Please use one of the options provided.")
+        amount_chosen = input(
+            "How much money would you like to withdraw? You may withdraw 20 EUR, 50 EUR or (O)ther amount.")
+
+    if amount_chosen == "O":
+        amount_chosen = input("Please enter the amount you would like to withdraw. ")
+        while not is_input_a_positive_float_with_a_max_of_two_decimal_places_check(amount_chosen):
+            amount_chosen = input("Please enter the amount you would like to withdraw. ")
+
+    amount_chosen = float(amount_chosen)
+
+    if check_if_balance_allows_withdrawal(amount_chosen):
+        update_balance(amount_chosen)
+        print("Here you go!")
+        time.sleep(1)
+
+    else:
+        print("You do not have enough money.")
+
+
+def withdraw_other_amount():
+    input_amount = input("Please enter the amount you would like to withdraw. ")
+    while not is_input_a_positive_float_with_a_max_of_two_decimal_places_check(input_amount):
+        input_amount = input("Please enter the amount you would like to withdraw. ")
+    return input_amount
+
+
+def update_balance(amount_withdrawn):
+    global balance
+    balance = balance - float(amount_withdrawn)
+
+
+def change_pin():
+    global pin
+    pin = set_pin()
+    print("You have successfully changed your PIN.")
+
+
+def check_if_balance_allows_withdrawal(amount_to_be_tested):
+    if balance - amount_to_be_tested > 0:
```

Comment:
this is redundant :)

you can just do something like this:
return balance - amount_to_be_tested > 0

make sure you understand why this works the same way.

Reply:
done. It works the same way because the expression "balance - amount_to_be_tested > 0" will evaluate to True when the balance is sufficient and False if it isn't, so it does exactly the same without the if statement. I changed it to >= because customers should be allowed to clean out their accounts completely